### PR TITLE
Connect to backend server and create recipe pages

### PR DESCRIPTION
**Pull request by dominiquecuevas**
1. Created a proxy to the flask backend routes
2. Recipes page with links to recipes
3. Recipe page with individual recipe
4. Update readme with cypress terminal instructions

image

https://github.com/athelia/full-spectrum-frontend/assets/56096225/f11a96e9-dc1b-49a7-92ed-2117a57eceb8

### CODE REVIEW

**Comment by dominiquecuevas** [comment]
pending merge of related api pull request [https://github.com/athelia/full-spectrum-api/pull/3](url)

File: cypress/videos/Header.cy.tsx.mp4
Comment (aileentran) [inline cypress/videos/Header.cy.tsx.mp4:1]: Qu'est-ce que c'est? Que esto?

**Review by athelia — CHANGES_REQUESTED** [review]
Add the cypress videos folder to .gitignore? Remove mp4 from commit

File: cypress/videos/Header.cy.tsx.mp4
Comment (dominiquecuevas) [reply cypress/videos/Header.cy.tsx.mp4:1]: deleted this, it was not needed

**Review by athelia — DISMISSED** [review]

File: pages/recipe/[id].tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+export const getStaticPaths: GetStaticPaths = async () => {
+  const recipes = await fetch(`http://${getHostname()}/api/recipes`)
+    .then((res) => {
+    if (res.ok) {
+      return res.json()
+    } else {
+      alert("something went wrong.")
+      throw res
```
Comment (aileentran) [inline pages/recipe/[id].tsx]: Refactor error: keep alert? add catch? pull out res.error or keep full res?

File: pages/recipe/[id].tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+export const getStaticProps: GetStaticProps = async ({ params }) => {
+  const recipe = await fetch(
+    `http://${getHostname()}/api/recipe/${params?.id}`
+  ).then((res) => {
+    if (res.ok) {
+      return res.json()
+    } else {
+      alert("something went wrong.")
+      throw res
```
Comment (aileentran) [inline pages/recipe/[id].tsx]: Potentially refactor - see above (line 78-ish).

File: pages/recipes.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import { getHostname } from '@/utilities'
+
+export const getStaticProps: GetStaticProps = async () => {
+  const recipes = await fetch(`http://${getHostname()}/api/recipes`)
+    .then(res => {
+      if (res.ok) {
+        return res.json()
+      } else {
+        alert('something went wrong.')
+        throw res
```
Comment (aileentran) [inline pages/recipes.tsx:12]: Potential refactor! See above x2!

**Review by aileentran — CHANGES_REQUESTED** [review]
Lookin' good for the most part! Just wanna check if you want to refactor your alerts in 3 diff places.

File: pages/recipe/[id].tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    </Card>
+  )
+}
+
+export const getStaticPaths: GetStaticPaths = async () => {
+  const recipes = await fetch(`http://${getHostname()}/api/recipes`)
+    .then((res) => {
+      if (res.ok) {
+        return res.json()
+      } else {
```
Comment (aileentran) [inline pages/recipe/[id].tsx]: Nice! I see that you refactored it. Hope it helps!

**Review by aileentran — DISMISSED** [review]
I see that you refactored the error catching in all 3 spots. Nice! Hope it helps!

File: README.md [hunk 5]
```diff
@@ -30,3 +30,10 @@ Open [http://localhost:3000](http://localhost:3000) in your browser
 ```bash
 npx cypress open
 ```
+alternatively run in the terminal
```
Comment (athelia) [inline README.md:33]: this is good, I should add testing guidelines for API too

File: utilities.tsx [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  let hostname = ''
+  if (env == 'development') {
+    hostname = '127.0.0.1:3000'
+  } else if (env == 'production') {
+    hostname = ''  // TODO: get production hostname when available
+  }
+  return hostname
+}
+
+export const stringToSlug = (text: String) => {
```
Comment (athelia) [inline utilities.tsx:12]: Can we talk about what this is for next time we get together? I see that it's used in the single recipe pages

File: pages/recipe/[recipeSlug]/[id].tsx [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    <Card>
+      <Card.Header>
+        <Card.Title>{name}</Card.Title>
+        <Card.Link
+          href={source}
+          target="_blank"
+          style={{ display: "block" }}
+        >
+          {url.hostname}
+          <svg
```
Comment (athelia) [inline pages/recipe/[recipeSlug]/[id].tsx]: Nit: can we wrap this SVG in a helper function so (1) it doesn't clutter up the Card component and (2) we can potentially use it other places?

**Review by athelia — DISMISSED** [review]
Overall looks great! One nit + a topic for next time we meet 👍🏼

File: utilities.tsx (hunk 6 above)
Comment (dominiquecuevas) [reply utilities.tsx:12]: To insert a human readable descriptive string to URL paths. 
It converts the recipe name to a URI compatible string. The regex removes spaces and then removes symbols

File: pages/recipe/[recipeSlug]/[id].tsx [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    <Card>
+      <Card.Header>
+        <Card.Title>{name}</Card.Title>
+        <Card.Link
+          href={source}
+          target="_blank"
+          style={{ display: "block" }}
+        >
+          {url.hostname}
+          <ExternalLinkSvg />
```
Comment (athelia) [inline pages/recipe/[recipeSlug]/[id].tsx:37]: So clean! Woooot

**Review by athelia — APPROVED** [review]
👍🏼

**Pull request merged**